docs/proposals/PROPOSAL_F.md

```diff
@@ -0,0 +1,190 @@
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/k8s.nfd.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-01-02T03:04:05Z"
+  }
+}
+```
+
+### Artifact
+
+The artifact is a simple json structure of named groups and within them, key value pairs. The groups can provide a higher level organization, such as types of features or even versions. This organization is up to the compatibility interest group. Here is an example organizing NFD annotations within extractor groups (a random selection):
+
+```json
+{
+  "compatibilities": {
+    "cpu": {
+      "cpu-hardware_multithreading": true,
+      "cpu-pstate.status": "passive",
+      "cpu-pstate.turbo": false
+    },
+    "kernel": {
+      "kernel-selinux.enabled": true,
+      "kernel-version.major": 4           
+    },
+    "memory": {
+      "memory-numa": true,
+      "memory-nv.dax": true
+    },
+    "network": {
+      "network-sriov.capable": true,
+      "network-sriov.configured": true
+    }
+}
+```
+
+A similar spec for an HPC-oriented container might look like the following:
+
+```json
+{
+  "compatibilities": {
+    "cpu": {
+      "target": "arm64",
+    },
+    "mpi": {
+      "implementation": "openmpi",
+      "version": "4.1.6"           
+    },
+    "gpu": {
+      "enabled": true
+    },
+    "os": {
+      "vendor": "rocky",
+      "version": "8"
+    }
+}
+```
+
+As stated previously, the granularity of metadata provided is up to the compatibility interest group. The degree to which each metadata attribute is used is up to the plugin or tool within the context of a specific use case.
+
+## Requirements
+
+Taken from [REQUIREMENTS](../REQUIREMENTS.md). Points that have:
+
+- `TOOL RESPONSIBILITY` would be possible by a custom tool.
+- `AGNOSTIC`: the proposal is agnostic to the specific point (can be decided by working group)
+
+### Image Author
+
+- [x] As an image author, I want to update compatibility independently without having to re-release and re-distribute my image.
+- [x] As an image author, I want to have the freedom to express any compatibility that is necessary for my container to run on the host.
+  - [x] Including conditional compatibility, such as a container running on AMD CPU and intel CPU having different requirements.
+  - [x] Including must have/nice to have compatibility.
+  - [x]  Matching finer grain platform definitions.
+- [x] As an image author, I want to use a provided tool to verify the compatibility spec I wrote against the schema.
```

Comment:
Does this proposal include the creation and maintenance of tools by OCI, or is it only a spec?

Reply:
It enables the writing of such tools.